README.md

```diff
@@ -167,23 +169,48 @@ Use [`func (tf *TF) Cmd(name string, args ...string) *exec.Cmd`](https://pkg.go.
+
+This is also provided via the convenience function `taskflow.VerboseParam(*taskflow.Taskflow)`, which needs to be explicitly called.
+The convenience function registers a boolean parameter `verbose`, short `v`, and assigns it to the `Taskflow` field.
+This parameter defaults to `false` by default, so as soon as this parameter is registered, verbose mode has to be explicitly enabled via the CLI.
+This then works similar to `go test -v`. 
+
+You can then also re-use the registered parameter to check if verbose mode was set within the task's command.
 
 ### Default task
 
 Default task can be assigned via `DefaultTask` field in [`type Taskflow`](https://pkg.go.dev/github.com/pellared/taskflow#Taskflow).
 
-When default task is set, then it is run if no task is provied via CLI.
+When default task is set, then it is run if no task is provided via CLI.
 
 ### Parameters
 
-The parameters can be set via CLI using the `key=val` syntax after CLI flags. For example, `go run ./build -v ci=true all` would run the `all` task with `ci` parameter set to `"true"` in verbose mode.
+The parameters can be set via CLI using flag syntax. Flags must be registered in "long form", and have an optional short form using only one letter.
+
+On the CLI, flags can be set in the following ways:
+* `--longName simple` - for simple single-word values
+* `--longName "value with blanks"`
+* `--longName="value with blanks"`
+* `--booleanParam` - setting boolean parameters implicitly to `true`
+* `-l simple` - using short form; Short form also allows assignment via equals.
+
+For example, `go run ./build -v --ci all` would run the `all` task with `ci` parameter set to `"true"` in verbose mode.
+
+Parameters must first be registered via [`func (*Taskflow) ConfigureValue`](https://pkg.go.dev/github.com/pellared/taskflow#Taskflow.ConfigureValue), or one of the provided primitive functions like `ConfigureString`.
+
+After registration, tasks need to specify which parameters they will read.
+Do this by assigning the `RegisteredParameter` from the registration result to the [`Task Parameters`](https://pkg.go.dev/github.com/pellared/taskflow#Task.Parameters) field.
+If a task tries to retrieve the value from an unregistered parameter, the task will fail. 
+
+With registration is done, the task's command can retrieve the parameter value using `Get(*TF)` from the respective `RegisteredParameter`, returned from the registration call during the task's `Command` execution.
 
-Default values can be assigned via `Params` field in [`type Taskflow`](https://pkg.go.dev/github.com/pellared/taskflow#Taskflow).
+See `examples/string-param/main.go` for a full parameter example.
 
-The task's command can get the parameters using [`func (*TF) Params`](https://pkg.go.dev/github.com/pellared/taskflow#TF.Params).
+Tasks cannot be registered with unknown parameters. Taskflow will fail execution if there are unused parameters.
```

Comment:
`Tasks cannot be registered with unknown parameters.` - not sure what you mean here. I guess that you mean that the parameters must be taken from the result of `Configure...` methods. If so I am not sure if mentioning it is needed. At least for me, it is confusing. You explained it pretty well in one of the previous paragraphs.